config/routes.rb

```diff
@@ -2622,17 +2622,16 @@
   get '/pictures/:basename' => 'picture#show', :basename => /[\da-zA-Z]+\.[\da-zA-Z]+/
 
   # Enablement for the REST API
-  # OPTIONS requests for REST API pre-flight checks
-
   # Semantic Versioning Regex for API, i.e. vMajor.minor.patch[-pre]
   apiver_regex = /v[\d]+(\.[\da-zA-Z]+)*(\-[\da-zA-Z]+)?/
 
-  match '/api/*path'   => 'api#handle_options_request', :via => [:options]
-  get '/api(/:version)'           => 'api#show',    :format => 'json', :version => apiver_regex
+  # OPTIONS requests for REST API pre-flight checks
+  match '/api/*path' => 'api#handle_options_request', :via => [:options]
 
-  get    '/api(/:version)/:collection(/:c_id(/:subcollection(/:s_id)))' => 'api#show',    :format => 'json', :version => apiver_regex
-  match  '/api(/:version)/:collection(/:c_id(/:subcollection(/:s_id)))' => 'api#update',  :format => 'json', :via => [:post, :put, :patch], :version => apiver_regex
-  delete '/api(/:version)/:collection(/:c_id(/:subcollection(/:s_id)))' => 'api#destroy', :format => 'json', :version => apiver_regex
+  get    '/api(/:version)(/*c_path)' => 'api#show',    :format => 'json', :version => apiver_regex
```

Comment:
I don't have the big picture here, but from an outside view I feel a bit unsure about this change. I'd like to be using Rails routing more, but the responsibility for routing is being moved further away from ActionDispatch and into the newly created `RequestAdapter` class. I'm wondering:

* If there's a way to append the arbitrary routing here while preserving the rules for every other endpoint?
* If it has to go, did you consider putting our custom routing into an object by itself that's responsible for only doing that?